tmva/tmva/src/DecisionTree.cxx

```diff
@@ -319,15 +412,280 @@ UInt_t TMVA::DecisionTree::BuildTree( const std::vector<const TMVA::Event*> & ev
    }
    else Log() << kFATAL << ":<BuildTree> eventsample Size == 0 " << Endl;
 
+
+   // sum up the totals
+   // sig and bkg for classification
+   // err and err2 for regression
+
+   // #### Set up prerequisite info for multithreading
+   UInt_t nPartitions = fNumCPUs;
+   auto seeds = ROOT::TSeqU(nPartitions);
```

Comment:
TThreadExecutor has mechanisms to implement chunking, you may want to check them. See it in action here: https://github.com/root-project/root/blob/master/math/mathcore/src/FitUtil.cxx#L370 where we have a function called setAutomaticChunking that makes a guess (based on observations, but almost arbitrary)of the number of chunks to take. It's better than chunking by number of logical processors in most of the cases.